tasks/get_current_archive_list/get_current_archive_list.py

```diff
@@ -0,0 +1,532 @@
+            )
+            # Update job status
+            LOGGER.debug(f"Posting successful status for job {job_id}.")
+            connection.execute(
+                update_job_sql(),
+                [
+                    {
+                        "id": job_id,
+                        "status_id": OrcaStatus.STAGED.value,
+                        "last_update": datetime.now(timezone.utc).isoformat(),
+                        "end_time": None,
+                        "error_message": None,
+                    }
+                ],
+            )
+    except Exception as sql_ex:
+        # Can't use f"" because of '{}' bug in CumulusLogger.
+        LOGGER.error(
+            "Error while processing job '{job_id}': {sql_ex}",
+            job_id=job_id,
+            sql_ex=sql_ex,
+        )
+        raise
+
+
+def add_metadata_to_gzip(report_bucket_name: str, gzip_key_path: str) -> None:
+    """
+    AWS does not add proper metadata to gzip files, which breaks aws_s3.table_import_from_s3
+    Must add manually.
+    Args:
+        report_bucket_name: The name of the bucket the csv is located in.
+        gzip_key_path: The path within the bucket to the gzip file that needs metadata updated.
+    """
+    s3 = boto3.resource("s3")
+    s3_object = s3.Object(report_bucket_name, gzip_key_path)
+    # Only add if needed.
+    if s3_object.content_encoding is None:
+        s3_object.copy_from(
+            CopySource={"Bucket": report_bucket_name, "Key": gzip_key_path},
+            Metadata=s3_object.metadata,
+            MetadataDirective="REPLACE",
+            ContentEncoding="gzip",
+        )
+
+
+# Keys indicate columns in the s3 inventory csv. Values indicate the corresponding column in orca.reconcile_s3_object
+column_mappings = {
+    "Bucket": "orca_archive_location text",
+    "Key": "key_path text",
+    "Size": "size_in_bytes bigint",
+    "LastModifiedDate": "last_update timestamptz",
+    "ETag": "etag text",
+    "StorageClass": "storage_class text",
+}
+
+
+def generate_temporary_s3_column_list(manifest_file_schema: str) -> str:
+    """
+    Creates a list of columns that matches the order of columns in the manifest.
+    Columns used by our database are given standardized names.
+    Args:
+        manifest_file_schema: An ordered CSV representing columns in the CSV the manifest points to.
+
+    Returns: A string representing SQL columns to create.
+        Columns required for import but unused by orca will be filled in with `junk` values.
+        For example, 'orca_archive_location text, key_path text, size_in_bytes bigint, last_update timestamptz, etag text, storage_class text, junk7 text, junk8 text, junk9 text, junk10 text, junk11 text, junk12 text, junk13 text, junk14 text'
+
+    """
+    column_index = 0
+    manifest_file_schema = manifest_file_schema.replace(" ", "")
+    columns_in_csv = manifest_file_schema.split(",")
+    columns_in_postgres = []
+    for column_in_csv in columns_in_csv:
```

Comment:
Since the Delete marker column will only occur in versioned buckets, I wonder if some logic should be added in to map it if it exists and add it if it doesn't with a default value of False.

Reply:
I'd vote on pushing this out until we can devote time to digging in. We don't actually know that the Delete marker will show up, I just think that is the most likely explanation for the unexplained presence on an example.